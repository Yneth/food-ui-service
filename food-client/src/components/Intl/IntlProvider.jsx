import React from 'react';
import PropTypes from 'prop-types';

class IntlProvider extends React.Component {

    getChildContext() {
        return {
            messages: this.props.messages,
            locale: this.props.locale,
        };
    }

    render() {
        const { children } = this.props;

        return children;
    }
}

IntlProvider.propTypes = {
    messages: PropTypes.objectOf(PropTypes.string),
    locale: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.arrayOf(PropTypes.element),
    ]).isRequired,
};

IntlProvider.defaultProps = {
    messages: {},
    locale: 'en',
};

IntlProvider.childContextTypes = {
    messages: PropTypes.objectOf(PropTypes.string),
    locale: PropTypes.string,
};

export default IntlProvider;
