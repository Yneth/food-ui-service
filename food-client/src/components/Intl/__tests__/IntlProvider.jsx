import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import IntlProvider from '../IntlProvider';

describe('IntlProvider', () => {
    it('should render children', () => {
        const tree = renderer.create(<IntlProvider><div>Provider</div></IntlProvider>);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should provide intl context', () => {
        const locale = 'fr';
        const messages = { hello: 'hi' };

        const tree = shallow(
            <IntlProvider locale={locale} messages={messages}><div>Provider</div></IntlProvider>,
        );

        expect(tree.instance().getChildContext()).toEqual({ locale, messages });
    });
});
