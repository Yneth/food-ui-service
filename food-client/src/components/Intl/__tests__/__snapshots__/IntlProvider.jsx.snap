// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntlProvider should render children 1`] = `
<div>
  Provider
</div>
`;
